django_short_urls/views.py

```diff
@@ -110,54 +111,72 @@ def main(request, path):
 
 
 @require_POST
-def new(request):
+def new(request):  # pylint: disable=too-many-branches
     '''
     Create a new short url based on the POST parameters
-    '''
 
-    if 'login' in request.REQUEST and 'api_key' in request.REQUEST:
-        login = request.REQUEST['login']
-        api_key = request.REQUEST['api_key']
+    Temporary pylint disable: OPS-4755 - once the temporary switches below have been removed,
+    remove the pylint disable above.
+    '''
+    user = None
+    if 'HTTP_AUTHORIZATION' in request.META:
+        method, auth = request.META['HTTP_AUTHORIZATION'].split(' ', 1)
+        if method.lower() == 'basic':
+            login, api_key = auth.strip().decode('base64').split(':', 1)
 
-        user = User.objects(login=login, api_key=api_key).first()
+            user = User.objects(login=login, api_key=api_key).first()
     else:
-        user = None
+        # Temporary – OPS-4755, allow specifying authentication parameters via request.REQUEST
+        try:
+            login, api_key = request.GET['login'], request.GET['api_key']
+        except KeyError:
+            getLogger('app').warning('/new called without credentials')
+            login, api_key = '', ''
+
+        statsd.increment('workforus.legacy_auth', tags=['user:' + login])
+        user = User.objects(login=login, api_key=api_key).first()
 
     if user is None:
         return response(status=HTTP_UNAUTHORIZED, message="Invalid credentials.")
 
-    params = {}
+    long_url = request.GET.get('long_url')
 
-    if 'long_url' in request.REQUEST:
-        params['long_url'] = request.REQUEST['long_url']
+    if long_url is None:
+        # Temporary – OPS-4755, allow specifying request parameters via request.POST
+        long_url = request.POST.get('long_url')
```

Comment:
It will, but I want to make sure I don't change any functionality until after we've released the new version of `work4us`. After discussing and double checking, this is probably not super useful after all.